templates/master.yaml

```diff
@@ -512,31 +411,26 @@ coreos:
     content: |
       [Unit]
       Description=etcd2
-      Requires=k8s-setup-network-env.service
-      After=k8s-setup-network-env.service
+      # TODO can we move this to a Wants in the decrypt-tls-certs.service ?
+      Requires=k8s-setup-network-env.service decrypt-tls-certs.service
+      After=k8s-setup-network-env.service decrypt-tls-certs.service
       Conflicts=etcd.service
       Wants=calico-node.service
+      StartLimitIntervalSec=0
 
       [Service]
       User=etcd
       Type=notify
       Restart=always
       RestartSec=0
       TimeoutStopSec=10
-      StartLimitIntervalSec=0
       LimitNOFILE=40000
       EnvironmentFile=/etc/network-environment
-      ExecStartPre=/usr/bin/etcdctl --endpoint=http://{{.Node.Hostname}}:2379 set /k8s/master/{{.Cluster.Etcd.Prefix}} '${DEFAULT_IPV4}'
```

Comment:
How do we want to do this then? I think the IP propagation and the endpoint updates have to be done as well on AWS. Is this right? I think we should use the same approaches across operators and only make differences where the underlying provider enforces this by its owm primitives. Like AWS has KMS. Fine. But in this case I think we have to update the endpoints with the actual IPs anyway: